packages/router/__tests__/router-test.ts

```diff
@@ -9323,41 +9434,92 @@ describe("a router", () => {
         });
       });
 
+      it("does not revalidate fetchers initiated from removed routes", async () => {
+        let t = setup({
+          routes: TASK_ROUTES,
+          initialEntries: ["/"],
+          hydrationData: { loaderData: { root: "ROOT", index: "INDEX" } },
+        });
+
+        let key = "key";
+
+        // Trigger a fetch from the index route
+        let A = await t.fetch("/tasks/1", key, "index");
+        await A.loaders.tasksId.resolve("TASKS");
+        expect(t.router.state.fetchers.get(key)).toMatchObject({
+          state: "idle",
+          data: "TASKS",
+        });
+
+        // Navigate such that the index route will be removed
+        let B = await t.navigate("/tasks", {
+          formMethod: "post",
+          formData: createFormData({}),
+        });
+
+        // Resolve the action
+        await B.actions.tasks.resolve("TASKS ACTION");
+
+        // Fetcher should remain in an idle state since it's calling route is
+        // being removed
+        expect(t.router.state.fetchers.get(key)).toMatchObject({
+          state: "idle",
+          data: "TASKS",
+        });
+
+        // Resolve navigation loaders
+        await B.loaders.root.resolve("ROOT*");
+        await B.loaders.tasks.resolve("TASKS LOADER");
+        expect(t.router.state.navigation.state).toBe("idle");
+        expect(t.router.state.location.pathname).toBe("/tasks");
+
+        // Fetcher never got called
+        expect(t.router.state.fetchers.get(key)).toMatchObject({
+          state: "idle",
+          data: "TASKS",
+        });
+      });
+
       it("cancels in-flight fetcher.loads on action submission and forces reload", async () => {
         let t = setup({
           routes: [
             {
-              id: "index",
```

Comment:
Wrapped all these in a parent route so the fetchers don't get skipped due to removed routes :)